### DISCUSSION TITLE
Possible to run two different test sets which are in the same folder

### DISCUSSION THREAD

**Discussion by JaapHaitsma** [orta/vscode-jest#1045] [Q&A] [answered]
Hi, in a project I'm in we have two test suites: unit-test and integration-tests

`jest.config.ts` has the configuration for unit tests
`jest.integration.config.ts` has the configuration for the integration test.

The tests of `jest.config.ts` are detected by jest extension, but the jest.integration.config.ts is not. Is there a way how this can done. I can change the commandline of jest to have it pickup the integration tests but then it does not pickup the unit tests

**Comment by connectdotz** [comment, marked as answer]
Hi, perfect timing. We just pre-released v6.0.0 with `virtualFolders` that specifically targeted these kinds of use cases. Please feel free to give it a try: https://github.com/jest-community/vscode-jest/blob/master/release-notes/release-note-v6.md#v600-pre-release

**Reply by JaapHaitsma** [reply]
@connectdotz trying it but it does not work for me. With this is my settings.json

`	"jest.virtualFolders": [
		{
			"name": "unit-tests",
			"jestCommandLine": "--config jest.config.ts",
			"autoRun": "off"
		}
	]`

I get the following output when trying to run the tests. When I remove the jest.virtualFolders from settings.json the test appear again

`[error] failed to retrieve test file list. TestExplorer might show incomplete test items
/bin/sh: --: invalid option
Usage:  /bin/sh [GNU long option] [option] ...
        /bin/sh [GNU long option] [option] script-file ...
GNU long options:
        --debug
        --debugger
        --dump-po-strings
        --dump-strings
        --help
        --init-file
        --login
        --noediting
        --noprofile
        --norc
        --posix
        --protected
        --rcfile
        --restricted
        --verbose
        --version
        --wordexp
Shell options:
        -irsD or -c command or -O shopt_option          (invocation only)
        -abefhkmnptuvxBCHP or -o option

> Test run exited at 7/17/2023, 11:48:02 AM <`

**Comment by connectdotz** [comment]
Oops, my bad. It looks like the document is wrong, The `"jestCommandLine"` is not complete, it's missing the actual jest command. 😓For example, if you usually start the test with "yarn test", then the setting should be: `"jestCommandLine": "yarn test --config jest.config.ts"`.  

Thanks for bringing it to our attention, I have corrected both README and release notes. Please give it a try again and let me know if you encounter any other issues.

**Reply by JaapHaitsma** [reply]
Thanks I got it working now.  Still a bug in your example code on the documentation page there should be no = after --config

**Reply by JaapHaitsma** [reply]
Also I use pnpm and pnpm test --config .... does not work. As pnpm does not expect an argument --config. So I went for npx jest --config .... Probably the safest option to do that because that should work for anybody right

**Reply by connectdotz** [reply]
hmm... the command line should work fine with or without "=". Are you sure the "=" sign caused a problem?

Regarding pnpm:  Like "npm", to append additional dashed arguments outside of the script, you need to give it an extra '--': `pnpm test -- --config xxxx`. (see [jestCommandLine](https://github.com/jest-community/vscode-jest/blob/master/README.md#jestcommandline))

Or you can just create a new script in the package.json, for example integration-test, that has the complete command:
```json
// package.json
"scripts": {
...
"test": "jest --config jest.config.js",
"integration-test": "jest --config jest.integration.config.ts"
}
```

Then in your virtualFolders, you can just say `"jestCommandLine": "pnpm integration-test --"`

Or you can bypass package.json script and run jest directly as you did with `npx`. Anyway, there are many ways to configure the [jestCommandLine](https://github.com/jest-community/vscode-jest/blob/master/README.md#jestcommandline). I am glad you got it working now! 👍
